pkg/alertmanager/operator.go

```diff
@@ -139,6 +144,8 @@ func New(ctx context.Context, restConfig *rest.Config, c operator.Config, logger
 
 		metrics:             operator.NewMetrics(r),
 		reconciliations:     &operator.ReconciliationTracker{},
+		eventRecorder:       operator.NewEventRecorder(client, OperatorName),
+		canEmitEvents:       canEmitEvents,
```

Comment:
(nit) we don't have to store the canEmitEvents bool in the struct.

```
o := &Operator{...}

var *eventsClient kubernetes.Interface
if canEmitEvents {
    eventsClient = client
}
o.eventRecorder = operator.NewEventRecorder(eventsClient, OperatorName)
````

Reply:
Ah nice idea, this would make the event recorder to send events to a "/dev/null sink" :)